### PR TITLE
docker: Fix test

### PR DESCRIPTION
**Pull request by mvollmer**
This works, but needs a `vm-install fedora-25` somwhere in testsuite-prepare

### CODE REVIEW

**Comment by petervo** [comment]
Pushed some commits to use the build image as the second machine in these tests.

File: test/common/vmmanage.py [hunk 1]
```diff
@@ -98,6 +98,20 @@ def only_install(image, skips=None, args=None, address=None):
         if started:
             machine.stop()
 
+# The Atomic variants can't build their own packages, so we build in
+# their non-Atomic siblings.  For example, fedora-atomic is build
```
Comment (dperpeet) [inline test/common/vmmanage.py]: `build` -> `built`

File: test/common/vmmanage.py [hunk 2]
```diff
@@ -126,6 +140,12 @@ def build_and_install(install_image, build_image, args):
                 build_and_maybe_install(build_image, do_install=False, skips=skips, args=args)
             if install_image:
                 only_install(install_image, skips, args=args)
+
+            # Atomics need a companion image for tests
+            if build_image and install_image and install_image in testinfra.ATOMIC_IMAGES:
+                skips.append("cockpit-ostree")
```
Comment (dperpeet) [inline test/common/vmmanage.py]: I don't quite understand why we don't want to install cockpit-ostree on Atomic systems...

File: test/verify/check-docker-storage [hunk 3]
```diff
@@ -84,19 +85,22 @@ class TestDockerStorage(MachineCase):
...
         # cockpit-ws, and one whose docker storage pool is managed.
         # We can't do that both on a single machine since resetting
         # the pool would kill the cockpit-ws container.
 
         if self.machine.atomic_image:
-            m = self.new_machine(machine_key='additional')
-            m.start()
-            m.wait_boot()
+            login_machine = self.new_machine(machine_key='login',
+                                             image=get_build_image(self.machine.image))
```
Comment (dperpeet) [inline test/verify/check-docker-storage]: I think we should have a comment here explaining why we use the `build` image as the `login` machine. For example:

```
# We want to use a non-Atomic machine as the login machine
# and the build machine that matches the current Atomic is suitable
```

File: test/common/vmmanage.py (hunk 2 above)
Comment (petervo) [inline test/common/vmmanage.py]: It's the build_image we don't install it on, not the install image. Note there are two install calls here.

File: test/common/vmmanage.py (hunk 1 above)
Comment (dperpeet) [inline test/common/vmmanage.py]: can be changed while merging also

File: test/verify/check-docker-storage (hunk 3 above)
Comment (petervo) [inline test/verify/check-docker-storage]: The main reason is speed, we need an image that has cockpit installed. We already have built rpms for that image so it makes it easy. The comment you have there is fine with me to add.

File: test/common/vmmanage.py (hunk 2 above)
Comment (dperpeet) [inline test/common/vmmanage.py]: right, thanks

**Pull request closed without merging**